sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java

```diff
@@ -0,0 +1,175 @@
+import com.azure.core.util.Context;
+import org.junit.jupiter.api.Test;
+import reactor.test.StepVerifier;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TextAnalyticsAsyncClientTest extends TextAnalyticsClientTestBase {
+    private TextAnalyticsAsyncClient client;
+
+    @Override
+    protected void beforeTest() {
+        client = clientSetup(httpPipeline -> new TextAnalyticsClientBuilder()
+            .endpoint(getEndPoint())
+            .pipeline(httpPipeline)
+            .buildAsyncClient());
+    }
+
+    /**
+     * Verify that we can get statistics on the collection result when given a batch input with options.
+     */
+    @Test
+    public void detectLanguagesBatchInputShowStatistics() {
+        detectLanguageShowStatisticsRunner((inputs, options) -> {
+            StepVerifier.create(client.detectBatchLanguagesWithResponse(inputs, options))
+                .assertNext(response -> validateBatchResult(response.getValue(), getExpectedBatchDetectedLanguages(), "Language"))
+                .verifyComplete();
+        });
+    }
+
+    /**
+     * Test Detect batch input languages.
+     */
+    @Test
+    public void detectLanguagesBatchInput() {
+        detectLanguageRunner((inputs) -> {
+            StepVerifier.create(client.detectBatchLanguages(inputs))
+                .assertNext(response -> validateBatchResult(response, getExpectedBatchDetectedLanguages(), "Language"))
+                .verifyComplete();
+        });
+    }
+
+    /**
+     * Test Detect batch languages for List of String input with country Hint.
+     */
+    @Test
+    public void detectLanguagesBatchListCountryHint() {
+        detectLanguagesCountryHintRunner((inputs, countryHint) -> {
+            StepVerifier.create(client.detectLanguagesWithResponse(inputs, countryHint))
+                .assertNext(response -> validateBatchResult(response.getValue(), getExpectedBatchDetectedLanguages(), "Language"))
+                .verifyComplete();
+        });
+    }
+
+    /**
+     * Test Detect batch languages for List of String input.
+     */
+    @Test
+    public void detectLanguagesBatchStringInput() {
+        detectLanguageStringInputRunner((inputs) -> {
+            StepVerifier.create(client.detectLanguages(inputs))
+                .assertNext(response -> validateBatchResult(response, getExpectedBatchDetectedLanguages(), "Language"))
+                .verifyComplete();
+        });
+    }
+
+    /**
+     * Verifies that a single DetectLanguageResult is returned for a text input to detectLanguages.
+     */
+    @Test
+    public void detectSingleTextLanguage() {
+        DetectedLanguage primaryLanguage = new DetectedLanguage().setName("English").setIso6391Name("en").setScore(1.0);
+        List<DetectedLanguage> expectedLanguageList = new ArrayList<>(Arrays.asList(primaryLanguage));
```

Comment:
`new ArrayList<>(Arrays.asList(` should just be `Arrays.asList(`. you may have to do a Ctrl F and replace all.